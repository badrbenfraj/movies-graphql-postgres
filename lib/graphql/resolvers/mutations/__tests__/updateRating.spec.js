const updateRating = require('../updateRating')
const { db } = require('../../../../adapters/db')
const getMovie = require('../../queries/movie')

jest.mock('../../../../adapters/db', () => ({
  db: {
    one: jest.fn().mockReturnValue('dbData'),
  },
  pgp: {
    helpers: {
      update: jest.fn().mockReturnValue('UPDATE.helper'),
    },
  },
}))

jest.mock('../../queries/movie', () => jest.fn())

describe('mutations/updateRating', () => {
  beforeEach(() => {
    db.one.mockClear()
  })

  it('should update a specific rating', () => {
    updateRating(null, { input: { movieId: '1', userId: 2, rating: 10 } })

    expect(db.one.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should get the movie', () => {
    updateRating(null, { input: { movieId: '1', userId: 2, rating: 10 } })

    expect(getMovie).toHaveBeenCalledWith(null, { id: '1' })
  })

  it('should return database data', async () => {
    getMovie.mockReturnValue(
      Promise.resolve({
        test: 'test',
      })
    )

    const response = await updateRating(null, {
      input: { movieId: '1', userId: 2, rating: 10 },
    })

    expect(response).toEqual({ test: 'test' })
  })

  it('should handle errors', async () => {
    db.one.mockImplementation(() => Promise.reject('nope'))

    try {
      await updateRating(null, {
        input: { movieId: '1', userId: 2, rating: 10 },
      })
    } catch (e) {
      expect(e.message).toMatch(/nope/)
    }
  })
})
