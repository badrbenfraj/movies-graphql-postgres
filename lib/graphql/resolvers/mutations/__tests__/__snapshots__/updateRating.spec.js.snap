// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations/updateRating should update a specific rating 1`] = `"UPDATE.helper WHERE user_id = 2 AND movie_id = 1 RETURNING *"`;
